### PR TITLE
feat: add create release action

### PR DESCRIPTION
**Pull request by vmaerten**
## What

The idea is to create automatically `release` in Github.
In this release, with the body automatically generated you will find : 
- tag name (date + prod or beta)
- All PR associated (who / what)
- Full changelog

We tried it on the collector: https://github.com/pbstck/collector/releases 

You can filter to find day or env release with the search input

## Why
It can be hard to find in Slack chan release, if / when / what has been released. Also, it's a shared Slack so we do not write every tech stuff

## And after ?
The goal is to modify https://github.com/pbstck/gh-follow-up to add one page to see all releases by time / by project / by env / etc
I also have an idea, which is to post a message in a Slack chan to avoid doing it manually (and forgetting it)

### CODE REVIEW

File: create-release/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+
+
+async function main() {
+    const octokit = new Octokit();
+    const [owner, repo] = process.env.GITHUB_REPOSITORY.split("/");
+    const tagSplitted = github.context.payload.ref.split('/');
+    if(tagSplitted[1]!== 'tags') {
+        throw new Error('It should be invoke only in a tag')
```
Comment (cforycki) [inline create-release/index.js:28]:
```suggestion
        throw new Error('It should be invoked only on a tag')
```

File: create-release/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        throw Error('Unknown environment')
+    }
+    return env
+}
+
+const getPreviousTag = (allTags, env) => {
+    return allTags.filter(t => {
+        const split = t.name.split('-');
+        return split[split.length-1] === env;
+    })[1];
```
Comment (cforycki) [inline create-release/index.js:18]:
```suggestion
    return allTags.filter(t => {
        const tagEnv = t.name.split('-').pop();
        return tagEnv === env;
    })[1];
```

File: create-release/index.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+
+
+async function main() {
+    const octokit = new Octokit();
+    const [owner, repo] = process.env.GITHUB_REPOSITORY.split("/");
+    const tagSplitted = github.context.payload.ref.split('/');
+    if(tagSplitted[1]!== 'tags') {
```
Comment (cforycki) [inline create-release/index.js:27]:
```suggestion
    if(tagSplitted[1] !== 'tags') {
```

File: create-release/index.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        throw new Error('It should be invoke only in a tag')
+    }
+    const env = getEnvironment(tagSplitted[2]);
+
+
```
Comment (cforycki) [inline create-release/index.js:32]:
```suggestion
```

File: create-release/index.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import { Octokit } from '@octokit/action'
+import * as github from '@actions/github'
+
+const getEnvironment = (tag) => {
+    let splitElement = tag.split('-');
+    const env = splitElement[splitElement.length-1]
```
Comment (cforycki) [inline create-release/index.js:7]:
```suggestion
    const env = tag.split('-').pop()
```

File: create-release/index.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    const previousTag = getPreviousTag(allTags.data, env);
+    console.log('Previous tag', previousTag.name)
+    console.log('tag',github.context.payload.ref)
+    const releaseNoteGenerated = await octokit.rest.repos.generateReleaseNotes({
+        owner,
+        repo,
+        tag_name: github.context.payload.ref,
+        previous_tag_name: previousTag.name
+    })
+    console.log(releaseNoteGenerated);
```
Comment (cforycki) [inline create-release/index.js:47]:
```suggestion
```

File: create-release/index.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        throw new Error('It should be invoke only in a tag')
+    }
+    const env = getEnvironment(tagSplitted[2]);
+
+
+    const allTags = await octokit.rest.repos.listTags({
+        owner,
+        repo,
+    });
+    console.log('All tags', allTags)
```
Comment (cforycki) [inline create-release/index.js:37]:
```suggestion
```

**Review by cforycki — APPROVED** [review]

**Review by lestrade-pbstck — APPROVED** [review]
Great addition !

**Comment by nicolas-raymond** [comment]
Are you the author of the script which doing this or not ?
If not, I don't find where it can from, and it's a good idea to be able to do it.

**Comment by nicolas-raymond** [comment]
> it's a shared Slack so we do not write every tech stuff

what about release messages posted on #techies channel ?

> I also have an idea, which is to post a message in a Slack chan to avoid doing it manually (and forgetting it)

well connected with my previous question, do you think on a specific channel or in #techies ?
Moreover, in my mind, if it's possible this automation should be mandatory :laughing:

**Comment by vmaerten** [comment]
> Are you the author of the script which doing this or not ? If not, I don't find where it can from, and it's a good idea to be able to do it.

I am. It's in the source https://github.com/pbstck/actions/pull/12/files#diff-73b8610c1890df31e13c31989da6510a54a48171312b4b20859940add55d199d

**Comment by vmaerten** [comment]
> > it's a shared Slack so we do not write every tech stuff
> 
> what about release messages posted on #techies channel ?
> 
> > I also have an idea, which is to post a message in a Slack chan to avoid doing it manually (and forgetting it)
> 
> well connected with my previous question, do you think on a specific channel or in #techies ? Moreover, in my mind, if it's possible this automation should be mandatory laughing

in #releases OR another channel but not techies (It should be discuss with everyone)
Well actually, I did not do everything because it was only a proposal and I wanted to do it step by step

**Review by nicolas-raymond — APPROVED** [review]

**Review by tvincent2 — APPROVED** [review]

**Review by sdeluce — APPROVED** [review]

**Pull request merged**